{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pybedtools\n",
    "import pandas as pd\n",
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/debapriyasinha/opt/anaconda3/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "track=pybedtools.BedTool('Example.gff')\n",
    "ref=pybedtools.BedTool('gencode.v28.annotation.gtf')\n",
    "gene = track.intersect(ref, wb=True)\n",
    "txtfile = gene.moveto('intersected.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/debapriyasinha/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py:4125: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().rename(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>attribute</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"; gene_type \"prote...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962047</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962047</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962355</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ref   start    stop  score strand  \\\n",
       "0  chr1  961771  962449   3.54      +   \n",
       "1  chr1  961771  962449   3.54      +   \n",
       "2  chr1  961826  962047   3.54      +   \n",
       "3  chr1  961826  962047   3.54      +   \n",
       "4  chr1  962355  962449   3.54      +   \n",
       "\n",
       "                                           attribute  \n",
       "0  gene_id \"ENSG00000187961.13\"; gene_type \"prote...  \n",
       "1  gene_id \"ENSG00000187961.13\"; transcript_id \"E...  \n",
       "2  gene_id \"ENSG00000187961.13\"; transcript_id \"E...  \n",
       "3  gene_id \"ENSG00000187961.13\"; transcript_id \"E...  \n",
       "4  gene_id \"ENSG00000187961.13\"; transcript_id \"E...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.read_csv('intersected.txt', names=['ref1', 'source1', 'method1', 'start1', 'stop1', 'score1', 'strand1', 'phase1', 'group1', 'ref2', 'source2', 'feature', 'start2', 'stop2', 'score2','strand2', 'frame', 'attribute'], sep='\\t')\n",
    "df.head()\n",
    "reqdf=df[['ref1','start1','stop1','score1','strand2','attribute']]\n",
    "reqdf.rename(columns={'ref1':'ref','start1':'start','stop1':'stop','score1':'score','strand2':'strand'}, inplace=True)\n",
    "reqdf.head()\n",
    "#[['ref','start','stop','score','strand','attribute']]\n",
    "#gene_type \"protein_coding\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>...</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000338591.7\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS30550.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000097875.3\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962047</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000338591.7\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"basic\"</td>\n",
       "      <td>tag \"appris_principal_1\"</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS30550.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000097875.3\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962047</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000338591.7\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"basic\"</td>\n",
       "      <td>tag \"appris_principal_1\"</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS30550.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000097875.3\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962355</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000338591.7\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"basic\"</td>\n",
       "      <td>tag \"appris_principal_1\"</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS30550.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000097875.3\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962355</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000338591.7\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"basic\"</td>\n",
       "      <td>tag \"appris_principal_1\"</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS30550.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000097875.3\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000622660.1\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962082</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000622660.1\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"basic\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962082</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000622660.1\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"basic\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040721.6\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000463212.1\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"retained_intron\"</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962449</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187961.13\"</td>\n",
       "      <td>transcript_id \"ENST00000463212.1\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>transcript_type \"retained_intron\"</td>\n",
       "      <td>...</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000097877.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1012424</td>\n",
       "      <td>1012804</td>\n",
       "      <td>1.05</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187608.8\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"ISG15\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040777.3\"</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1012424</td>\n",
       "      <td>1012804</td>\n",
       "      <td>1.05</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187608.8\"</td>\n",
       "      <td>transcript_id \"ENST00000624697.3\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"ISG15\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040777.3\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000479384.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1012424</td>\n",
       "      <td>1012804</td>\n",
       "      <td>1.05</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"ENSG00000187608.8\"</td>\n",
       "      <td>transcript_id \"ENST00000624652.1\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"ISG15\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>havana_gene \"OTTHUMG00000040777.3\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000479385.2\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000078808.16\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001812.6\"</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000078808.16\"</td>\n",
       "      <td>transcript_id \"ENST00000360001.10\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS30553.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001812.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000005064.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000078808.16\"</td>\n",
       "      <td>transcript_id \"ENST00000465727.5\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>transcript_type \"nonsense_mediated_decay\"</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000078808.16\"</td>\n",
       "      <td>transcript_id \"ENST00000263741.11\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS12.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001812.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000005070.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000078808.16\"</td>\n",
       "      <td>transcript_id \"ENST00000459994.2\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>transcript_type \"processed_transcript\"</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1246514</td>\n",
       "      <td>1246571</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"ENSG00000184163.3\"</td>\n",
       "      <td>transcript_id \"ENST00000330388.2\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_name \"C1QTNF12\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>...</td>\n",
       "      <td>tag \"basic\"</td>\n",
       "      <td>tag \"appris_principal_1\"</td>\n",
       "      <td>tag \"CCDS\"</td>\n",
       "      <td>ccdsid \"CCDS30554.1\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001412.6\"</td>\n",
       "      <td>havana_transcript \"OTTHUMT00000004080.4\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     ref    start     stop  score strand                             0  \\\n",
       "0   chr1   961771   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "1   chr1   961771   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "2   chr1   961826   962047   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "3   chr1   961826   962047   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "4   chr1   962355   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "5   chr1   962355   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "6   chr1   961771   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "7   chr1   962082   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "8   chr1   962082   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "9   chr1   961771   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "10  chr1   961826   962449   3.54      +  gene_id \"ENSG00000187961.13\"   \n",
       "11  chr1  1012424  1012804   1.05      +   gene_id \"ENSG00000187608.8\"   \n",
       "12  chr1  1012424  1012804   1.05      +   gene_id \"ENSG00000187608.8\"   \n",
       "13  chr1  1012424  1012804   1.05      +   gene_id \"ENSG00000187608.8\"   \n",
       "16  chr1  1231002  1231015   4.09      -  gene_id \"ENSG00000078808.16\"   \n",
       "17  chr1  1231002  1231015   4.09      -  gene_id \"ENSG00000078808.16\"   \n",
       "18  chr1  1231002  1231015   4.09      -  gene_id \"ENSG00000078808.16\"   \n",
       "19  chr1  1231002  1231015   4.09      -  gene_id \"ENSG00000078808.16\"   \n",
       "20  chr1  1231002  1231015   4.09      -  gene_id \"ENSG00000078808.16\"   \n",
       "21  chr1  1246514  1246571   0.22      -   gene_id \"ENSG00000184163.3\"   \n",
       "\n",
       "                                      1                            2  \\\n",
       "0            gene_type \"protein_coding\"           gene_name \"KLHL17\"   \n",
       "1     transcript_id \"ENST00000338591.7\"   gene_type \"protein_coding\"   \n",
       "2     transcript_id \"ENST00000338591.7\"   gene_type \"protein_coding\"   \n",
       "3     transcript_id \"ENST00000338591.7\"   gene_type \"protein_coding\"   \n",
       "4     transcript_id \"ENST00000338591.7\"   gene_type \"protein_coding\"   \n",
       "5     transcript_id \"ENST00000338591.7\"   gene_type \"protein_coding\"   \n",
       "6     transcript_id \"ENST00000622660.1\"   gene_type \"protein_coding\"   \n",
       "7     transcript_id \"ENST00000622660.1\"   gene_type \"protein_coding\"   \n",
       "8     transcript_id \"ENST00000622660.1\"   gene_type \"protein_coding\"   \n",
       "9     transcript_id \"ENST00000463212.1\"   gene_type \"protein_coding\"   \n",
       "10    transcript_id \"ENST00000463212.1\"   gene_type \"protein_coding\"   \n",
       "11           gene_type \"protein_coding\"            gene_name \"ISG15\"   \n",
       "12    transcript_id \"ENST00000624697.3\"   gene_type \"protein_coding\"   \n",
       "13    transcript_id \"ENST00000624652.1\"   gene_type \"protein_coding\"   \n",
       "16           gene_type \"protein_coding\"             gene_name \"SDF4\"   \n",
       "17   transcript_id \"ENST00000360001.10\"   gene_type \"protein_coding\"   \n",
       "18    transcript_id \"ENST00000465727.5\"   gene_type \"protein_coding\"   \n",
       "19   transcript_id \"ENST00000263741.11\"   gene_type \"protein_coding\"   \n",
       "20    transcript_id \"ENST00000459994.2\"   gene_type \"protein_coding\"   \n",
       "21    transcript_id \"ENST00000330388.2\"   gene_type \"protein_coding\"   \n",
       "\n",
       "                        3                                           4  ...  \\\n",
       "0                 level 2          havana_gene \"OTTHUMG00000040721.6\"  ...   \n",
       "1      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "2      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "3      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "4      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "5      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "6      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "7      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "8      gene_name \"KLHL17\"            transcript_type \"protein_coding\"  ...   \n",
       "9      gene_name \"KLHL17\"           transcript_type \"retained_intron\"  ...   \n",
       "10     gene_name \"KLHL17\"           transcript_type \"retained_intron\"  ...   \n",
       "11                level 2          havana_gene \"OTTHUMG00000040777.3\"  ...   \n",
       "12      gene_name \"ISG15\"            transcript_type \"protein_coding\"  ...   \n",
       "13      gene_name \"ISG15\"            transcript_type \"protein_coding\"  ...   \n",
       "16                level 2          havana_gene \"OTTHUMG00000001812.6\"  ...   \n",
       "17       gene_name \"SDF4\"            transcript_type \"protein_coding\"  ...   \n",
       "18       gene_name \"SDF4\"   transcript_type \"nonsense_mediated_decay\"  ...   \n",
       "19       gene_name \"SDF4\"            transcript_type \"protein_coding\"  ...   \n",
       "20       gene_name \"SDF4\"      transcript_type \"processed_transcript\"  ...   \n",
       "21   gene_name \"C1QTNF12\"            transcript_type \"protein_coding\"  ...   \n",
       "\n",
       "                                           11  \\\n",
       "0                                        None   \n",
       "1                                  tag \"CCDS\"   \n",
       "2                                 tag \"basic\"   \n",
       "3                                 tag \"basic\"   \n",
       "4                                 tag \"basic\"   \n",
       "5                                 tag \"basic\"   \n",
       "6                                               \n",
       "7                                 tag \"basic\"   \n",
       "8                                 tag \"basic\"   \n",
       "9                                        None   \n",
       "10   havana_transcript \"OTTHUMT00000097877.1\"   \n",
       "11                                       None   \n",
       "12         havana_gene \"OTTHUMG00000040777.3\"   \n",
       "13         havana_gene \"OTTHUMG00000040777.3\"   \n",
       "16                                       None   \n",
       "17                                 tag \"CCDS\"   \n",
       "18                                              \n",
       "19                                 tag \"CCDS\"   \n",
       "20                                              \n",
       "21                                tag \"basic\"   \n",
       "\n",
       "                                           12  \\\n",
       "0                                        None   \n",
       "1                        ccdsid \"CCDS30550.1\"   \n",
       "2                    tag \"appris_principal_1\"   \n",
       "3                    tag \"appris_principal_1\"   \n",
       "4                    tag \"appris_principal_1\"   \n",
       "5                    tag \"appris_principal_1\"   \n",
       "6                                        None   \n",
       "7          havana_gene \"OTTHUMG00000040721.6\"   \n",
       "8          havana_gene \"OTTHUMG00000040721.6\"   \n",
       "9                                        None   \n",
       "10                                              \n",
       "11                                       None   \n",
       "12   havana_transcript \"OTTHUMT00000479384.1\"   \n",
       "13   havana_transcript \"OTTHUMT00000479385.2\"   \n",
       "16                                       None   \n",
       "17                       ccdsid \"CCDS30553.1\"   \n",
       "18                                       None   \n",
       "19                          ccdsid \"CCDS12.1\"   \n",
       "20                                       None   \n",
       "21                   tag \"appris_principal_1\"   \n",
       "\n",
       "                                     13  \\\n",
       "0                                  None   \n",
       "1    havana_gene \"OTTHUMG00000040721.6\"   \n",
       "2                            tag \"CCDS\"   \n",
       "3                            tag \"CCDS\"   \n",
       "4                            tag \"CCDS\"   \n",
       "5                            tag \"CCDS\"   \n",
       "6                                  None   \n",
       "7                                         \n",
       "8                                         \n",
       "9                                  None   \n",
       "10                                 None   \n",
       "11                                 None   \n",
       "12                                        \n",
       "13                                        \n",
       "16                                 None   \n",
       "17   havana_gene \"OTTHUMG00000001812.6\"   \n",
       "18                                 None   \n",
       "19   havana_gene \"OTTHUMG00000001812.6\"   \n",
       "20                                 None   \n",
       "21                           tag \"CCDS\"   \n",
       "\n",
       "                                           14  \\\n",
       "0                                        None   \n",
       "1    havana_transcript \"OTTHUMT00000097875.3\"   \n",
       "2                        ccdsid \"CCDS30550.1\"   \n",
       "3                        ccdsid \"CCDS30550.1\"   \n",
       "4                        ccdsid \"CCDS30550.1\"   \n",
       "5                        ccdsid \"CCDS30550.1\"   \n",
       "6                                        None   \n",
       "7                                        None   \n",
       "8                                        None   \n",
       "9                                        None   \n",
       "10                                       None   \n",
       "11                                       None   \n",
       "12                                       None   \n",
       "13                                       None   \n",
       "16                                       None   \n",
       "17   havana_transcript \"OTTHUMT00000005064.1\"   \n",
       "18                                       None   \n",
       "19   havana_transcript \"OTTHUMT00000005070.1\"   \n",
       "20                                       None   \n",
       "21                       ccdsid \"CCDS30554.1\"   \n",
       "\n",
       "                                     15  \\\n",
       "0                                  None   \n",
       "1                                         \n",
       "2    havana_gene \"OTTHUMG00000040721.6\"   \n",
       "3    havana_gene \"OTTHUMG00000040721.6\"   \n",
       "4    havana_gene \"OTTHUMG00000040721.6\"   \n",
       "5    havana_gene \"OTTHUMG00000040721.6\"   \n",
       "6                                  None   \n",
       "7                                  None   \n",
       "8                                  None   \n",
       "9                                  None   \n",
       "10                                 None   \n",
       "11                                 None   \n",
       "12                                 None   \n",
       "13                                 None   \n",
       "16                                 None   \n",
       "17                                        \n",
       "18                                 None   \n",
       "19                                        \n",
       "20                                 None   \n",
       "21   havana_gene \"OTTHUMG00000001412.6\"   \n",
       "\n",
       "                                           16    17    18    19    20  \n",
       "0                                        None  None  None  None  None  \n",
       "1                                        None  None  None  None  None  \n",
       "2    havana_transcript \"OTTHUMT00000097875.3\"        None  None  None  \n",
       "3    havana_transcript \"OTTHUMT00000097875.3\"        None  None  None  \n",
       "4    havana_transcript \"OTTHUMT00000097875.3\"        None  None  None  \n",
       "5    havana_transcript \"OTTHUMT00000097875.3\"        None  None  None  \n",
       "6                                        None  None  None  None  None  \n",
       "7                                        None  None  None  None  None  \n",
       "8                                        None  None  None  None  None  \n",
       "9                                        None  None  None  None  None  \n",
       "10                                       None  None  None  None  None  \n",
       "11                                       None  None  None  None  None  \n",
       "12                                       None  None  None  None  None  \n",
       "13                                       None  None  None  None  None  \n",
       "16                                       None  None  None  None  None  \n",
       "17                                       None  None  None  None  None  \n",
       "18                                       None  None  None  None  None  \n",
       "19                                       None  None  None  None  None  \n",
       "20                                       None  None  None  None  None  \n",
       "21   havana_transcript \"OTTHUMT00000004080.4\"        None  None  None  \n",
       "\n",
       "[20 rows x 26 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#select the rows that contains the string 'gene_type \"protein_coding\"'\n",
    "reqdf = reqdf[reqdf['attribute'].str.contains('gene_type \"protein_coding\"')]\n",
    "#split the string attribute into columns\n",
    "reqdf = reqdf.join(reqdf['attribute'].str.split(';', expand=True))\n",
    "reqdf = reqdf.drop('attribute', 1)\n",
    "\n",
    "reqdf.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>...</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"KLHL17\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"ISG15\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"ISG15\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"ISG15\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"SDF4\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"C1QTNF12\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"C1QTNF12\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"C1QTNF12\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"C1QTNF12\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"ACAP3\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gene_name \"ACAP3\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    ref  start  stop  score strand    0    1                      2    4    5  \\\n",
       "0   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "1   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "2   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "3   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "4   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "5   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "6   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "7   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "8   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "9   NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "10  NaN    NaN   NaN    NaN    NaN  NaN  NaN     gene_name \"KLHL17\"  NaN  NaN   \n",
       "11  NaN    NaN   NaN    NaN    NaN  NaN  NaN      gene_name \"ISG15\"  NaN  NaN   \n",
       "12  NaN    NaN   NaN    NaN    NaN  NaN  NaN      gene_name \"ISG15\"  NaN  NaN   \n",
       "13  NaN    NaN   NaN    NaN    NaN  NaN  NaN      gene_name \"ISG15\"  NaN  NaN   \n",
       "16  NaN    NaN   NaN    NaN    NaN  NaN  NaN       gene_name \"SDF4\"  NaN  NaN   \n",
       "17  NaN    NaN   NaN    NaN    NaN  NaN  NaN       gene_name \"SDF4\"  NaN  NaN   \n",
       "18  NaN    NaN   NaN    NaN    NaN  NaN  NaN       gene_name \"SDF4\"  NaN  NaN   \n",
       "19  NaN    NaN   NaN    NaN    NaN  NaN  NaN       gene_name \"SDF4\"  NaN  NaN   \n",
       "20  NaN    NaN   NaN    NaN    NaN  NaN  NaN       gene_name \"SDF4\"  NaN  NaN   \n",
       "21  NaN    NaN   NaN    NaN    NaN  NaN  NaN   gene_name \"C1QTNF12\"  NaN  NaN   \n",
       "22  NaN    NaN   NaN    NaN    NaN  NaN  NaN   gene_name \"C1QTNF12\"  NaN  NaN   \n",
       "23  NaN    NaN   NaN    NaN    NaN  NaN  NaN   gene_name \"C1QTNF12\"  NaN  NaN   \n",
       "24  NaN    NaN   NaN    NaN    NaN  NaN  NaN   gene_name \"C1QTNF12\"  NaN  NaN   \n",
       "27  NaN    NaN   NaN    NaN    NaN  NaN  NaN      gene_name \"ACAP3\"  NaN  NaN   \n",
       "28  NaN    NaN   NaN    NaN    NaN  NaN  NaN      gene_name \"ACAP3\"  NaN  NaN   \n",
       "\n",
       "    ...   11   12   13   14   15   16   17   18   19   20  \n",
       "0   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "1   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "2   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "3   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "4   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "5   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "6   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "7   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "8   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "9   ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "10  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "11  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "12  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "13  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "16  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "17  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "18  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "19  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "20  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "21  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "22  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "23  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "24  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "27  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "28  ...  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  \n",
       "\n",
       "[25 rows x 25 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#select the column that contains string \"gene_name\"\n",
    "list(reqdf.columns)\n",
    "selected_cols = list(range(0, 21))\n",
    "#reqdf[selected_cols]\n",
    "newdf=reqdf[reqdf[selected_cols].apply(lambda x: x.str.contains('gene_name'))]\n",
    "newdf[2].update(newdf.pop(3))\n",
    "newdf.isnull().sum()\n",
    "newdf.head(25)\n",
    "#newdf=newdf.drop.isnull()\n",
    "#newdf.head(30)\n",
    "#reqdf[reqdf.select_dtypes(include='object').apply(lambda x: x.str.contains('gene_name \"SDF4\"')).all(axis=1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = [reqdf[\"ref\"], reqdf['start'], reqdf['stop'], newdf[2], reqdf['score'], reqdf['strand']]\n",
    "headers = ['ref','start','stop','name','score','strand']\n",
    "df3 = pd.concat(data, axis=1, keys=headers)\n",
    "df3 = df3.join(df3['name'].str.split('\"', expand=True))\n",
    "df4 = [df3[\"ref\"], df3['start'], df3['stop'], df3[1], df3['score'], df3['strand']]\n",
    "headers = ['ref','start','stop','name','score','strand']\n",
    "finaldf = pd.concat(df4, axis=1, keys=headers)\n",
    "finaldf.head(20)\n",
    "finaldf.to_csv('output.bed', sep='\\t', index=False)\n",
    "finaldf.to_csv('output.bed', sep='\\t', quoting=csv.QUOTE_NONE, escapechar='\"', index=False)\n",
    "#cant use bedtools merge since that will unify many separate peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962047</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962047</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962355</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962355</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962082</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962082</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961771</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961826</td>\n",
       "      <td>962449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>3.54</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1012424</td>\n",
       "      <td>1012804</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>1.05</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1012424</td>\n",
       "      <td>1012804</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>1.05</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1012424</td>\n",
       "      <td>1012804</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>1.05</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1231002</td>\n",
       "      <td>1231015</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>4.09</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1246514</td>\n",
       "      <td>1246571</td>\n",
       "      <td>C1QTNF12</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1246514</td>\n",
       "      <td>1246571</td>\n",
       "      <td>C1QTNF12</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1245723</td>\n",
       "      <td>1246571</td>\n",
       "      <td>C1QTNF12</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1245723</td>\n",
       "      <td>1246571</td>\n",
       "      <td>C1QTNF12</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1298567</td>\n",
       "      <td>1298679</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1298567</td>\n",
       "      <td>1298679</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     ref    start     stop      name  score strand\n",
       "0   chr1   961771   962449    KLHL17   3.54      +\n",
       "1   chr1   961771   962449    KLHL17   3.54      +\n",
       "2   chr1   961826   962047    KLHL17   3.54      +\n",
       "3   chr1   961826   962047    KLHL17   3.54      +\n",
       "4   chr1   962355   962449    KLHL17   3.54      +\n",
       "5   chr1   962355   962449    KLHL17   3.54      +\n",
       "6   chr1   961771   962449    KLHL17   3.54      +\n",
       "7   chr1   962082   962449    KLHL17   3.54      +\n",
       "8   chr1   962082   962449    KLHL17   3.54      +\n",
       "9   chr1   961771   962449    KLHL17   3.54      +\n",
       "10  chr1   961826   962449    KLHL17   3.54      +\n",
       "11  chr1  1012424  1012804     ISG15   1.05      +\n",
       "12  chr1  1012424  1012804     ISG15   1.05      +\n",
       "13  chr1  1012424  1012804     ISG15   1.05      +\n",
       "16  chr1  1231002  1231015      SDF4   4.09      -\n",
       "17  chr1  1231002  1231015      SDF4   4.09      -\n",
       "18  chr1  1231002  1231015      SDF4   4.09      -\n",
       "19  chr1  1231002  1231015      SDF4   4.09      -\n",
       "20  chr1  1231002  1231015      SDF4   4.09      -\n",
       "21  chr1  1246514  1246571  C1QTNF12   0.22      -\n",
       "22  chr1  1246514  1246571  C1QTNF12   0.22      -\n",
       "23  chr1  1245723  1246571  C1QTNF12   0.22      -\n",
       "24  chr1  1245723  1246571  C1QTNF12   0.22      -\n",
       "27  chr1  1298567  1298679     ACAP3   0.26      -\n",
       "28  chr1  1298567  1298679     ACAP3   0.26      -"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "finaldf.head(25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACAP3</td>\n",
       "      <td>0.26</td>\n",
       "      <td>1297956</td>\n",
       "      <td>1298958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACAP3</td>\n",
       "      <td>1.40</td>\n",
       "      <td>1309345</td>\n",
       "      <td>1309567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAP3</td>\n",
       "      <td>1.55</td>\n",
       "      <td>1302261</td>\n",
       "      <td>1304260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACBD3</td>\n",
       "      <td>0.18</td>\n",
       "      <td>226186066</td>\n",
       "      <td>226186730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACOT11</td>\n",
       "      <td>1.02</td>\n",
       "      <td>54622952</td>\n",
       "      <td>54624556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.69</td>\n",
       "      <td>1736593</td>\n",
       "      <td>1737476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ADIPOR1</td>\n",
       "      <td>0.96</td>\n",
       "      <td>202957647</td>\n",
       "      <td>202958572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AFAP1L2</td>\n",
       "      <td>0.41</td>\n",
       "      <td>114403486</td>\n",
       "      <td>114404756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AHCTF1</td>\n",
       "      <td>1.12</td>\n",
       "      <td>246930425</td>\n",
       "      <td>246931978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AK2</td>\n",
       "      <td>1.70</td>\n",
       "      <td>33080114</td>\n",
       "      <td>33080996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AK4</td>\n",
       "      <td>1.16</td>\n",
       "      <td>65147549</td>\n",
       "      <td>65148632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ALG14</td>\n",
       "      <td>3.96</td>\n",
       "      <td>95071946</td>\n",
       "      <td>95072945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ALPL</td>\n",
       "      <td>2.57</td>\n",
       "      <td>21509372</td>\n",
       "      <td>21510372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AMY2A</td>\n",
       "      <td>3.70</td>\n",
       "      <td>103616811</td>\n",
       "      <td>103617114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ANK3</td>\n",
       "      <td>1.08</td>\n",
       "      <td>60140222</td>\n",
       "      <td>60140639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ANXA7</td>\n",
       "      <td>3.20</td>\n",
       "      <td>73413068</td>\n",
       "      <td>73413068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AQP10</td>\n",
       "      <td>1.61</td>\n",
       "      <td>154324554</td>\n",
       "      <td>154325325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ARHGAP21</td>\n",
       "      <td>2.46</td>\n",
       "      <td>24722645</td>\n",
       "      <td>24723668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ARHGEF19</td>\n",
       "      <td>0.89</td>\n",
       "      <td>16206324</td>\n",
       "      <td>16208011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ARID4B</td>\n",
       "      <td>1.73</td>\n",
       "      <td>235327785</td>\n",
       "      <td>235328219</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        name  score      start       stop\n",
       "0      ACAP3   0.26    1297956    1298958\n",
       "1      ACAP3   1.40    1309345    1309567\n",
       "2      ACAP3   1.55    1302261    1304260\n",
       "3      ACBD3   0.18  226186066  226186730\n",
       "4     ACOT11   1.02   54622952   54624556\n",
       "5     ADARB2   0.69    1736593    1737476\n",
       "6    ADIPOR1   0.96  202957647  202958572\n",
       "7    AFAP1L2   0.41  114403486  114404756\n",
       "8     AHCTF1   1.12  246930425  246931978\n",
       "9        AK2   1.70   33080114   33080996\n",
       "10       AK4   1.16   65147549   65148632\n",
       "11     ALG14   3.96   95071946   95072945\n",
       "12      ALPL   2.57   21509372   21510372\n",
       "13     AMY2A   3.70  103616811  103617114\n",
       "14      ANK3   1.08   60140222   60140639\n",
       "15     ANXA7   3.20   73413068   73413068\n",
       "16     AQP10   1.61  154324554  154325325\n",
       "17  ARHGAP21   2.46   24722645   24723668\n",
       "18  ARHGEF19   0.89   16206324   16208011\n",
       "19    ARID4B   1.73  235327785  235328219"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combdf = finaldf.groupby(['name','score']).agg({'start':'min', 'stop':'max'}).reset_index()\n",
    "combdf.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref</th>\n",
       "      <th>name</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>C1QTNF12</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     ref      name strand\n",
       "0   chr1    KLHL17      +\n",
       "11  chr1     ISG15      +\n",
       "16  chr1      SDF4      -\n",
       "21  chr1  C1QTNF12      -\n",
       "27  chr1     ACAP3      -"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "remdata = [finaldf[\"ref\"], finaldf['name'], finaldf['strand']]\n",
    "headers = ['ref','name','strand']\n",
    "remcols = pd.concat(remdata, axis=1, keys=headers)\n",
    "remcols = remcols.drop_duplicates(subset='name', keep=\"first\")\n",
    "remcols.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACAP3</td>\n",
       "      <td>0.26</td>\n",
       "      <td>1297956</td>\n",
       "      <td>1298958</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACAP3</td>\n",
       "      <td>1.40</td>\n",
       "      <td>1309345</td>\n",
       "      <td>1309567</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAP3</td>\n",
       "      <td>1.55</td>\n",
       "      <td>1302261</td>\n",
       "      <td>1304260</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACBD3</td>\n",
       "      <td>0.18</td>\n",
       "      <td>226186066</td>\n",
       "      <td>226186730</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACOT11</td>\n",
       "      <td>1.02</td>\n",
       "      <td>54622952</td>\n",
       "      <td>54624556</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.69</td>\n",
       "      <td>1736593</td>\n",
       "      <td>1737476</td>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ADIPOR1</td>\n",
       "      <td>0.96</td>\n",
       "      <td>202957647</td>\n",
       "      <td>202958572</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AFAP1L2</td>\n",
       "      <td>0.41</td>\n",
       "      <td>114403486</td>\n",
       "      <td>114404756</td>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AHCTF1</td>\n",
       "      <td>1.12</td>\n",
       "      <td>246930425</td>\n",
       "      <td>246931978</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AK2</td>\n",
       "      <td>1.70</td>\n",
       "      <td>33080114</td>\n",
       "      <td>33080996</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AK4</td>\n",
       "      <td>1.16</td>\n",
       "      <td>65147549</td>\n",
       "      <td>65148632</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ALG14</td>\n",
       "      <td>3.96</td>\n",
       "      <td>95071946</td>\n",
       "      <td>95072945</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ALPL</td>\n",
       "      <td>2.57</td>\n",
       "      <td>21509372</td>\n",
       "      <td>21510372</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AMY2A</td>\n",
       "      <td>3.70</td>\n",
       "      <td>103616811</td>\n",
       "      <td>103617114</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ANK3</td>\n",
       "      <td>1.08</td>\n",
       "      <td>60140222</td>\n",
       "      <td>60140639</td>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ANXA7</td>\n",
       "      <td>3.20</td>\n",
       "      <td>73413068</td>\n",
       "      <td>73413068</td>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AQP10</td>\n",
       "      <td>1.61</td>\n",
       "      <td>154324554</td>\n",
       "      <td>154325325</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ARHGAP21</td>\n",
       "      <td>2.46</td>\n",
       "      <td>24722645</td>\n",
       "      <td>24723668</td>\n",
       "      <td>chr10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ARHGEF19</td>\n",
       "      <td>0.89</td>\n",
       "      <td>16206324</td>\n",
       "      <td>16208011</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ARID4B</td>\n",
       "      <td>1.73</td>\n",
       "      <td>235327785</td>\n",
       "      <td>235328219</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        name  score      start       stop    ref strand\n",
       "0      ACAP3   0.26    1297956    1298958   chr1      -\n",
       "1      ACAP3   1.40    1309345    1309567   chr1      -\n",
       "2      ACAP3   1.55    1302261    1304260   chr1      -\n",
       "3      ACBD3   0.18  226186066  226186730   chr1      -\n",
       "4     ACOT11   1.02   54622952   54624556   chr1      +\n",
       "5     ADARB2   0.69    1736593    1737476  chr10      -\n",
       "6    ADIPOR1   0.96  202957647  202958572   chr1      -\n",
       "7    AFAP1L2   0.41  114403486  114404756  chr10      -\n",
       "8     AHCTF1   1.12  246930425  246931978   chr1      -\n",
       "9        AK2   1.70   33080114   33080996   chr1      -\n",
       "10       AK4   1.16   65147549   65148632   chr1      +\n",
       "11     ALG14   3.96   95071946   95072945   chr1      -\n",
       "12      ALPL   2.57   21509372   21510372   chr1      +\n",
       "13     AMY2A   3.70  103616811  103617114   chr1      +\n",
       "14      ANK3   1.08   60140222   60140639  chr10      -\n",
       "15     ANXA7   3.20   73413068   73413068  chr10      -\n",
       "16     AQP10   1.61  154324554  154325325   chr1      +\n",
       "17  ARHGAP21   2.46   24722645   24723668  chr10      -\n",
       "18  ARHGEF19   0.89   16206324   16208011   chr1      -\n",
       "19    ARID4B   1.73  235327785  235328219   chr1      -"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "finalcomb = pd.merge(combdf,remcols,how = \"inner\", on = \"name\")\n",
    "finalcomb.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = [finalcomb[\"ref\"], finalcomb['start'], finalcomb['stop'], finalcomb['name'], finalcomb['score'], finalcomb['strand']]\n",
    "#headers = ['ref','start','stop','name','score','strand']\n",
    "foroutput = pd.concat(data, axis=1)\n",
    "foroutput.head(20)\n",
    "foroutput = foroutput.rename(columns=foroutput.iloc[0]).drop(foroutput.index[0])\n",
    "foroutput.to_csv('afterfiltration.bed', sep='\\t')\n",
    "foroutput.to_csv('afterfiltration.bed', sep='\\t', quoting=csv.QUOTE_NONE, escapechar='\"', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/debapriyasinha/opt/anaconda3/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<BedTool(afterfiltration_sorted.bed)>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a=pybedtools.BedTool('afterfiltration.bed').sort()\n",
    "a.moveto('afterfiltration_sorted.bed')\n",
    "#tosort=a.sort()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
